x-pack/plugins/telemetry/index.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resolve } from 'path';
+import JoiNamespace from 'joi';
+import { Server } from 'hapi';
+import { CoreSetup, PluginInitializerContext } from 'src/core/server';
+import { i18n } from '@kbn/i18n';
+import mappings from './mappings.json';
+import { CONFIG_TELEMETRY, getConfigTelemetryDesc } from './common/constants';
+import { plugin } from './server';
+import { REPORT_INTERVAL_MS } from './common/constants';
+import { checkDeprecated } from './common/check_deprecated';
+
+import {
+  createLocalizationUsageCollector,
+  createTelemetryUsageCollector,
+} from './server/collectors';
+
+export const telemetry = (kibana: any) => {
+  return new kibana.Plugin({
+    id: 'telemetry',
+    configPrefix: 'xpack.telemetry',
+    publicDir: resolve(__dirname, 'public'),
+    require: ['elasticsearch'],
+    config(Joi: typeof JoiNamespace) {
+      return Joi.object({
+        enabled: Joi.boolean().default(true),
+        // `config` is used internally and not intended to be set
+        config: Joi.string().default(Joi.ref('$defaultConfigPath')),
+        url: Joi.when('$dev', {
+          is: true,
+          then: Joi.string().default('https://telemetry-staging.elastic.co/xpack/v2/send'),
```

Comment:
I think I would prefer the API path to be a variable here, that way when/if we change to v3 you only have to change it in one place.

Reply:
makes sense :+1: